utils/wasm-gen/src/tests.rs

```diff
@@ -157,6 +163,169 @@ fn injecting_addresses_works() {
+            "syscall: {}",
+            syscall.to_str()
+        );
+
+        // Assert that syscall results will be ignored.
+        let termination_reason =
+            execute_wasm_with_syscall_injected(syscall, true, params_config, initial_memory_write);
+
+        assert_eq!(
+            termination_reason,
+            TerminationReason::Actor(ActorTerminationReason::Success),
+            "syscall: {}",
+            syscall.to_str()
+        );
+    }
+}
+
+#[derive(Clone)]
+struct MemoryWrite {
+    offset: u32,
+    content: Vec<u8>,
+}
+
+fn get_params_for_syscall_to_fail(
+    syscall: &SysCallName,
+) -> (SysCallsParamsConfig, Option<MemoryWrite>) {
+    let memory_write = match *syscall {
+        SysCallName::PayProgramRent => Some(MemoryWrite {
+            offset: 0,
+            content: vec![255; WASM_PAGE_SIZE],
+        }),
+        _ => None,
+    };
+
+    (
+        SysCallsParamsConfig::all_constant_value(i32::MAX as i64),
+        memory_write,
+    )
+}
+
+fn execute_wasm_with_syscall_injected(
+    syscall: SysCallName,
+    ignore_fallible_errors: bool,
+    params_config: SysCallsParamsConfig,
+    initial_memory_write: Option<MemoryWrite>,
+) -> TerminationReason {
+    use gear_backend_common::{BackendReport, Environment};
+    use gear_backend_sandbox::SandboxEnvironment;
+    use gear_core::message::{ContextSettings, DispatchKind, IncomingDispatch, MessageContext};
+    use gear_core_processor::{ProcessorContext, ProcessorExternalities};
+
+    const INITIAL_PAGES: u16 = 1;
+    const INJECTED_SYSCALLS: u32 = 8;
+
+    // We create Unstructured from zeroes here as we just need any
+    let buf = vec![0; UNSTRUCTURED_SIZE];
+    let mut unstructured = Unstructured::new(&buf);
+
+    let mut injection_amounts = SysCallsInjectionAmounts::all_never();
+    injection_amounts.set(syscall, INJECTED_SYSCALLS, INJECTED_SYSCALLS);
+
+    let gear_config = (
+        GearWasmGeneratorConfigBuilder::new()
+            .with_memory_config(MemoryPagesConfig {
+                initial_size: INITIAL_PAGES as u32,
+                ..MemoryPagesConfig::default()
+            })
+            .with_sys_calls_config(
+                SysCallsConfigBuilder::new(injection_amounts)
+                    .with_params_config(params_config)
+                    .set_ignore_fallible_syscall_errors(ignore_fallible_errors)
+                    .build(),
+            )
+            .with_entry_points_config(EntryPointsSet::Init)
+            .build(),
+        SelectableParams {
+            call_indirect_enabled: false,
+            allowed_instructions: vec![],
+            max_instructions: 0,
+            min_funcs: 1,
+            max_funcs: 1,
+        },
+    );
+
+    let module = generate_gear_program_module(&mut unstructured, gear_config)
+        .expect("failed wasm generation");
+
+    let module = gear_wasm_instrument::inject(
+        module,
+        &gear_wasm_instrument::rules::CustomConstantCostRules::new(0, 0, 0),
```

Comment:
```suggestion
        &CustomConstantCostRules::new(0, 0, 0),
```

full diff: https://gist.github.com/StackOverflowExcept1on/bd009da4be071e8d64d7064cde5fe5a0